rclcpp_components/cmake/rclcpp_components_register_node.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# usage: rclcpp_components_register_node(
+#              <target> PLUGIN <component> EXECUTABLE <node>)
+#
+# Register an rclcpp component with the ament
+# resource index and create an executable.
+#
+# :param target: the shared library target
+# :type target: string
+# :param ARGS: the plugin name and node name for executable
+# :type ARGS: list of strings
+# :param component: the component plugin name
+# :type component: string
+# :param node: the node's executable name
+# :type node: string
+#
+macro(rclcpp_components_register_node target)
+  cmake_parse_arguments(
+    ARGS
+    ""
+    "PLUGIN;EXECUTABLE"
+    ""
+    ${ARGN})
+  set(component ${ARGS_PLUGIN})
+  set(node ${ARGS_EXECUTABLE})
+  _rclcpp_components_register_package_hook()
+  if(WIN32)
+    set(_path "bin")
+  else()
+    set(_path "lib")
+  endif()
+  set(_RCLCPP_COMPONENTS__NODES
+    "${_RCLCPP_COMPONENTS__NODES}${component};${_path}/$<TARGET_FILE_NAME:${target}>\n")
+  set(CLASS_NAME ${component})
+  set(build_dir ${PROJECT_BINARY_DIR}/../rclcpp_components) 
```

Comment:
This is fragile. It assumes that `rclcpp_components` is at a specific location relative to the build directory of the project invoking `rclcpp_components_register_node`.  Instead, we should export a variable pointing to the install directory that can be used here, e.g. `rclcpp_components_NODE_MAIN_TEMPLATE`.

This can be achieved with a config file or generator expressions, I believe.